### PR TITLE
[Refactor] add pillow backend for ColorJitter related functions

### PR DESCRIPTION
**Pull request by fangyixiao18**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

the training acc drops in mmselfsup if using cv2 backend

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/image/photometric.py [hunk 1]
```diff
@@ -110,22 +115,41 @@ def adjust_color(img, alpha=1, beta=None, gamma=0):
...
-        # is needed.
-        colored_img = np.clip(colored_img, 0, 255)
-    return colored_img
+    if backend is None:
+        backend = imread_backend
+    if backend not in ['cv2', 'pillow']:
+        raise ValueError(f'backend: {backend} is not supported.'
+                         f"Supported backends are 'cv2', 'pillow'")
+
+    if backend == 'pillow':
```
Comment (zhouzaida) [inline mmcv/image/photometric.py:132]: Should we also test whether the results of two backends are the same?

**Comment by fangyixiao18** [comment]
pytest tests/test_image/test_photometric.py
bymasIq36T

**Review by zhouzaida — APPROVED** [review]

File: mmcv/image/photometric.py [hunk 2]
```diff
@@ -449,23 +511,51 @@ def adjust_hue(img: np.ndarray, hue_factor: float) -> np.ndarray:
             HSV space in positive and negative direction respectively.
             0 means no shift. Therefore, both -0.5 and 0.5 will give an image
             with complementary colors while 0 gives the original image.
+        backend (str | None): The image processing backend type. Options are
+            `cv2`, `pillow`, `None`. If backend is None, the global
+            imread_backend specified by ``mmcv.use_backend()`` will be used.
```
Comment (HAOCHENYE) [inline mmcv/image/photometric.py:516]:
```suggestion
            ``imread_backend`` specified by ``mmcv.use_backend()`` will be used.
```

File: mmcv/image/photometric.py [hunk 3]
```diff
@@ -219,20 +261,38 @@ def adjust_contrast(img, factor=1.):
     Args:
         img (ndarray): Image to be contrasted. BGR order.
         factor (float): Same as :func:`mmcv.adjust_brightness`.
+        backend (str | None): The image processing backend type. Options are
+            `cv2`, `pillow`, `None`. If backend is None, the global
+            imread_backend specified by ``mmcv.use_backend()`` will be used.
```
Comment (HAOCHENYE) [inline mmcv/image/photometric.py:266]: ``imread_backend``

File: mmcv/image/photometric.py [hunk 4]
```diff
@@ -110,22 +115,41 @@ def adjust_color(img, alpha=1, beta=None, gamma=0):
             If None, it's assigned the value (1 - `alpha`).
         gamma (int | float): Scalar added to each sum.
             Same as :func:`cv2.addWeighted`. Default 0.
+        backend (str | None): The image processing backend type. Options are
+            `cv2`, `pillow`, `None`. If backend is None, the global
+            imread_backend specified by ``mmcv.use_backend()`` will be used.
```
Comment (HAOCHENYE) [inline mmcv/image/photometric.py:120]:
```suggestion
            ``imread_backend`` specified by ``mmcv.use_backend()`` will be used.
```

File: mmcv/image/photometric.py [hunk 5]
```diff
@@ -110,22 +115,41 @@ def adjust_color(img, alpha=1, beta=None, gamma=0):
...
+        if beta is None:
+            beta = 1 - alpha
+        colored_img = cv2.addWeighted(img, alpha, gray_img, beta, gamma)
+        if not colored_img.dtype == np.uint8:
+            # Note when the dtype of `img` is not the default `np.uint8`
+            # (e.g. np.float32), the value in `colored_img` got from cv2
+            # is not guaranteed to be in range [0, 255], so here clip
+            # is needed.
+            colored_img = np.clip(colored_img, 0, 255)
+        return colored_img
```
Comment (HAOCHENYE) [inline mmcv/image/photometric.py:152]: Why not call ``colored_img.astype(img.dtype)`` just like below.

**Review by HAOCHENYE — APPROVED** [review]

**Pull request merged**